### ISSUE TITLE
Upgrade from 0.3.6 to 0.3.7 breaking simple authentication

### ISSUE THREAD

**Issue by dplunk** [venth/aws-adfs#49]
I just upgraded my module from 0.3.6 to 0.3.7 and am now receiving the following error during authentication. 

```
(adfs) thecomputer:~ <myusername>$ aws-adfs login --adfs-host=mydomain.com --region=us-east-1
2017-07-14 12:51:16,509 [authenticator authenticator.py:authenticate] [80410-MainProcess] [140735207329792-MainThread] - ERROR: Cannot extract saml assertion. Second factor authentication failed?
Username: <myusername>@mydomain.com
Password:
Traceback (most recent call last):
  File "/Users/<myusername>/adfs/bin/aws-adfs", line 11, in <module>
    sys.exit(cli())
  File "/Users/<myusername>/adfs/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/<myusername>/adfs/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/<myusername>/adfs/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/<myusername>/adfs/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/<myusername>/adfs/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/<myusername>/adfs/lib/python2.7/site-packages/aws_adfs/login.py", line 91, in login
    principal_roles, assertion, aws_session_duration = authenticator.authenticate(config, username, password)
  File "/Users/<myusername>/adfs/lib/python2.7/site-packages/aws_adfs/authenticator.py", line 38, in authenticate
    principal_roles)
  File "/Users/<myusername>/adfs/lib/python2.7/site-packages/aws_adfs/authenticator.py", line 57, in _aggregate_roles_by_account_alias
    if account_aliases[account_no] not in aggregated_accounts:
KeyError: '214329012423'
(adfs) thecomputer:~ <myusername>$
````

Python version 2.7.10 on Mac OS X El Capitan 10.11.4
Pip modules installed:
```
asn1crypto==0.22.0
aws-adfs==0.3.7
awscli==1.11.105
boto3==1.4.4
botocore==1.5.82
certifi==2017.4.17
cffi==1.10.0
chardet==3.0.4
click==6.7
colorama==0.3.7
configparser==3.5.0
cryptography==1.9
docutils==0.13.1
enum34==1.1.6
futures==3.1.1
idna==2.5
ipaddress==1.0.18
jmespath==0.9.3
lxml==3.8.0
pyasn1==0.2.3
pycparser==2.18
pyOpenSSL==17.1.0
python-dateutil==2.6.1
PyYAML==3.12
requests==2.18.1
rsa==3.4.2
s3transfer==0.1.10
six==1.10.0
urllib3==1.21.1
```

**Comment by venth** [comment]
Do you have environment variable AWS_PROFILE set?

**Comment by dplunk** [comment]
@venth no, there's no AWS_PROFILE set. The only AWS_* env is AWS_DEFAULT_REGION.

**Comment by venth** [comment]
Would you like to send me results of `aws-adfs --verbose login ....` command via e-mail? My email address is: artur(dot)krysiak(dot)warszawa(at)gmail.com

I found the cause and I need HTML results to see what I missed during parsing.

**Comment by dplunk** [comment]
@venth A sanitized output has been sent your way, thank you!

**Comment by asceresini** [comment]
@venth would you like any more outputs? Otherwise I will just watch and wait for a fix. I've downgraded back to known working version of 0.3.3 as an interim measure.

**Comment by venth** [comment]
Now, I'm completely occupied by my son. That's why fixes are postponed... :(

**Comment by venth** [comment]
Patch provided with version: 0.3.8

**Comment by anthoneous** [comment]
This issue doesn't appear to be fixed in version 0.3.8

`2017-07-27 17:51:10,754 [authenticator authenticator.py:authenticate] [90346-MainProcess] [140736902673344-MainThread] - ERROR: Cannot extract saml assertion. Second factor authentication failed?
Username [user@user.com]:
Password:
Sending request for authentication
Waiting for additional authentication
Going for aws roles
Traceback (most recent call last):
  File "/usr/local/bin/aws-adfs", line 11, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/aws_adfs/login.py", line 91, in login
    principal_roles, assertion, aws_session_duration = authenticator.authenticate(config, username, password)
  File "/usr/local/lib/python2.7/site-packages/aws_adfs/authenticator.py", line 38, in authenticate
    principal_roles)
  File "/usr/local/lib/python2.7/site-packages/aws_adfs/authenticator.py", line 57, in _aggregate_roles_by_account_alias
    if account_aliases[account_no] not in aggregated_accounts:
KeyError: '############'
05:51 PM:-=@MAC: >`

**Comment by venth** [comment]
So, the cause I found seems to be different. I found out that for the failing account number its alias was not provided and the results in html looked like:
`<div class="saml-account-name">Account: 1234567890</div>` instead of:
`<div class="saml-account-name">Account: Account-Awesome-Alias (1234567890)</div>`
It seems that, handling this specific situation didn't help. I'll fix it as follows:
* in case there cannot be account alias extracted, I'll use simply account number as it's alias.

**Comment by venth** [comment]
Afore mentioned change delivered with version 0.3.9

**Comment by asceresini** [comment]
Hi @venth,

The issue seems to be resolved, however, I am receiving this output prior to entering username:

`2017-07-29 08:00:19,403 [authenticator authenticator.py:authenticate] [29856-MainProcess] [140736811815872-MainThread] - ERROR: Cannot extract saml assertion. Second factor authentication failed?`

**Comment by venth** [comment]
The messages, you haven seen, are caused by the authentication attempt. For the first time aws-adfs tries to authenticate against adfs server based on the information stored in a profile chosen by --profile switch. The assumption is that you're already authenticated and want to extend aws session. If the attempt fails, regular authentication is performed.

**Comment by anthoneous** [comment]
@dplunk @venth I can confirm this bug is resolved in 0.3.9.

**Comment by venth** [comment]
Thanks @anthoneous. I close the issue then.
